ipa-core/src/protocol/hybrid/breakdown_reveal.rs

```diff
@@ -127,30 +60,29 @@ where
 ///     record. This is currently ensured by the serial operation of the aggregation
 ///     protocol (i.e. by not using `seq_join`).
 #[tracing::instrument(name = "breakdown_reveal_aggregation", skip_all, fields(total = attributed_values.len()))]
-pub async fn breakdown_reveal_aggregation<C, BK, TV, HV, const B: usize>(
+pub async fn breakdown_reveal_aggregation<C, BK, V, HV, const B: usize>(
     ctx: C,
-    attributed_values: Vec<SecretSharedAttributionOutputs<BK, TV>>,
+    attributed_values: Vec<AggregateableHybridReport<BK, V>>,
     padding_params: &PaddingParameters,
 ) -> Result<BitDecomposed<Replicated<Boolean, B>>, Error>
 where
     C: UpgradableContext + Shuffle,
     Boolean: FieldSimd<B>,
     Replicated<Boolean, B>: BooleanProtocols<DZKPUpgraded<C>, B>,
-    BK: BreakdownKey<B>,
+    BK: BooleanArray + U128Conversions,
     Replicated<BK>: Reveal<DZKPUpgraded<C>, Output = <BK as Vectorizable<1>>::Array>,
-    TV: BooleanArray + U128Conversions,
+    V: BooleanArray + U128Conversions,
     HV: BooleanArray + U128Conversions,
     BitDecomposed<Replicated<Boolean, B>>:
-        for<'a> TransposeFrom<&'a [Replicated<TV>; B], Error = Infallible>,
+        for<'a> TransposeFrom<&'a [Replicated<V>; B], Error = Infallible>,
 {
     // Apply DP padding for Breakdown Reveal Aggregation
-    let attributed_values_padded =
-        apply_dp_padding::<_, AttributionOutputs<Replicated<BK>, Replicated<TV>>, B>(
-            ctx.narrow(&Step::PaddingDp),
-            attributed_values,
-            padding_params,
-        )
-        .await?;
+    let attributed_values_padded = apply_dp_padding::<_, AggregateableHybridReport<BK, V>, B>(
+        ctx.narrow(&Step::PaddingDp),
+        attributed_values,
+        padding_params,
+    )
+    .await?;
 
     let attributions = ctx
```

Comment:
are we using sharded shuffle here?

Reply:
good catch. added `ShardedContext` to the type bounds for the context, so we should be now.